sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -7,227 +7,182 @@
-
-    # Test dense data
-    svd_a_10_de = TruncatedSVD(10, algorithm="arpack")
-    svd_r_10_de = TruncatedSVD(10, algorithm="randomized", random_state=42)
-    svd_a_20_de = TruncatedSVD(20, algorithm="arpack")
-    svd_r_20_de = TruncatedSVD(20, algorithm="randomized", random_state=42)
-    X_trans_a_10_de = svd_a_10_de.fit_transform(X.toarray())
-    X_trans_r_10_de = svd_r_10_de.fit_transform(X.toarray())
-    X_trans_a_20_de = svd_a_20_de.fit_transform(X.toarray())
-    X_trans_r_20_de = svd_r_20_de.fit_transform(X.toarray())
-
-    # helper arrays for tests below
-    svds = (svd_a_10_sp, svd_r_10_sp, svd_a_20_sp, svd_r_20_sp, svd_a_10_de,
-            svd_r_10_de, svd_a_20_de, svd_r_20_de)
-    svds_trans = (
-        (svd_a_10_sp, X_trans_a_10_sp),
-        (svd_r_10_sp, X_trans_r_10_sp),
-        (svd_a_20_sp, X_trans_a_20_sp),
-        (svd_r_20_sp, X_trans_r_20_sp),
-        (svd_a_10_de, X_trans_a_10_de),
-        (svd_r_10_de, X_trans_r_10_de),
-        (svd_a_20_de, X_trans_a_20_de),
-        (svd_r_20_de, X_trans_r_20_de),
-    )
-    svds_10_v_20 = (
-        (svd_a_10_sp, svd_a_20_sp),
-        (svd_r_10_sp, svd_r_20_sp),
-        (svd_a_10_de, svd_a_20_de),
-        (svd_r_10_de, svd_r_20_de),
-    )
-    svds_sparse_v_dense = (
-        (svd_a_10_sp, svd_a_10_de),
-        (svd_a_20_sp, svd_a_20_de),
-        (svd_r_10_sp, svd_r_10_de),
-        (svd_r_20_sp, svd_r_20_de),
-    )
+    assert Xtrans.shape == (n_samples, tsvd.n_components)
 
-    # Assert the 1st component is equal
-    for svd_10, svd_20 in svds_10_v_20:
-        assert_array_almost_equal(
-            svd_10.explained_variance_ratio_,
-            svd_20.explained_variance_ratio_[:10],
-            decimal=5,
-        )
-
-    # Assert that 20 components has higher explained variance than 10
-    for svd_10, svd_20 in svds_10_v_20:
-        assert_greater(
-            svd_20.explained_variance_ratio_.sum(),
-            svd_10.explained_variance_ratio_.sum(),
-        )
 
+@pytest.mark.parametrize('kind', ('dense', 'sparse'))
+@pytest.mark.parametrize('n_components', [10, 20])
+@pytest.mark.parametrize('solver', SVD_SOLVERS)
+def test_explained_variance(X_sparse, kind, n_components, solver):
+    X = X_sparse if kind == 'sparse' else X_sparse.toarray()
+    svd = TruncatedSVD(n_components, algorithm=solver)
+    X_tr = svd.fit_transform(X)
     # Assert that all the values are greater than 0
-    for svd in svds:
-        assert_array_less(0.0, svd.explained_variance_ratio_)
+    assert_array_less(0.0, svd.explained_variance_ratio_)
 
     # Assert that total explained variance is less than 1
-    for svd in svds:
-        assert_array_less(svd.explained_variance_ratio_.sum(), 1.0)
-
-    # Compare sparse vs. dense
-    for svd_sparse, svd_dense in svds_sparse_v_dense:
-        assert_array_almost_equal(svd_sparse.explained_variance_ratio_,
-                                  svd_dense.explained_variance_ratio_)
```

Comment:
This was removed as it is indirectly tested with checking that `explained_variance_ratio_` for both dense and sparse is equal to the expected value below.